### ISSUE TITLE
[doc] Create a simple online Ruleset Editor

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4296]
This is basically a Javascript app that uses the generated rule catalog in JSON format (#4295) and presents a simple UI:
* A list of available rules
* Filters to narrow down the list of rules (language, name, category, tags, ...)
* A list of selected rules
* A button to export the selected rules as a ruleset XML file

In a second phase, when a rule in the "selected rules" is selected, one can configure
the properties and priority.

In a third phase, we could provide support for file-filters (includes, excludes, ...)

In a fourth phase, we could support importing an existing ruleset so that one can
continue modifying the ruleset with that editor.

I guess, for PMD 7.0, the first phase would already be value to get experience
how useful this is. Note, that this editor doesn't replace the rule reference
documentation, since the editor won't display all information (like example configurations,
examples, full description, ...).

**Comment by jsotuyod** [comment]
@delanym no, the designer is intended to help build new rules (ie: test XPath expressions, explore sample code ASTs, create test cases, etc.). This is purely a tool to generate a ruleset XML from already existing (standard) rules.

**Comment by delanym** [comment]
@jsotuyod ok. But help me understand what the end game is here. Are there plans for a third tool to edit existing rulesets? I hope not.
